### PR TITLE
Makefile: don't globally export GOOS=linux

### LINKED ISSUE #220: 'go run' uses GOOS=linux for local tools, even on non-linux platforms
https://github.com/operator-framework/operator-controller/blob/4ea83e6fbe666af5023d4e43f34110ac2cd27e24/Makefile#L115
makes GOOS 'sticky' for follow-on operations, and since it is the first dependency for the 'run' target, it /might/ be run first and carry that value over to other targets.

### PR DESCRIPTION
**Pull request by joelanford**
# Description
Fixes #220

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,26 +120,8 @@ wait:
...
-.PHONY: docker-buildx
-docker-buildx: test ## Build and push docker image for the manager for cross-platform support
-	# copy existing Dockerfile and insert --platform=${BUILDPLATFORM} into Dockerfile.cross, and preserve the original Dockerfile
-	sed -e '1 s/\(^FROM\)/FROM --platform=\$$\{BUILDPLATFORM\}/; t' -e ' 1,// s//FROM --platform=\$$\{BUILDPLATFORM\}/' Dockerfile > Dockerfile.cross
-	- docker buildx create --name project-v3-builder
-	docker buildx use project-v3-builder
-	- docker buildx build --push --platform=$(PLATFORMS) --tag ${IMG} -f Dockerfile.cross .
-	- docker buildx rm project-v3-builder
-	rm Dockerfile.cross
+docker-build: build-linux ## Build docker image with the operator-controller.
```
Comment (tmshort) [inline Makefile:123]: Should the comment here indicate GOOS/GOARCH behavior?

**Review by tmshort — APPROVED** [review]
Looks good, with one minor question.

File: Makefile (hunk 1 above)
Comment (joelanford) [reply Makefile:123]: Yeah, good idea.

Comment (joelanford) [reply Makefile]: Updated.

**Comment by tmshort** [comment]
Still looks good, waiting for CI

**Pull request merged**